src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1716,6 +1716,11 @@ private bool pInvokeMarshalingRequired(CORINFO_METHOD_STRUCT_* handle, CORINFO_S
             if (method.IsRawPInvoke())
                 return false;
 
+            // Stub is required to trigger precise static constructor
+            TypeDesc owningType = method.OwningType;
+            if (_compilation.HasLazyStaticConstructor(owningType) && !((MetadataType)owningType).IsBeforeFieldInit)
```

Comment:
Adding this check into `IsPInvokeStubRequired` would require passing Compilation into new places, so I have added it here instead.